File: server/java/spring/user/src/main/java/xyz/hoper/user/client/GrpcClientManager.java
package xyz.hoper.user.client;


import java.util.concurrent.TimeUnit;
import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;
import org.springframework.stereotype.Component;

@Component
public class GrpcClientManager {
    ManagedChannel getChannel(String host,Integer port) {
        return ManagedChannelBuilder.forAddress(host, port)
                .usePlaintext()
                .disableRetry()
                .idleTimeout(5, TimeUnit.SECONDS)
                .build();
    }
}

File: server/kotlin/spring/user/src/main/kotlin/xyz/hoper/user/client/GrpcClientManager.kt
package xyz.hoper.user.client

import org.springframework.stereotype.Component
import java.util.concurrent.TimeUnit
import io.grpc.ManagedChannel
import io.grpc.ManagedChannelBuilder

@Component
class GrpcClientManager {
    fun getChannel(host: String?, port: Int): ManagedChannel? {
        return ManagedChannelBuilder.forAddress(host, port)
                .usePlaintext()
                .disableRetry()
                .idleTimeout(5, TimeUnit.SECONDS)
                .build()
    }
}